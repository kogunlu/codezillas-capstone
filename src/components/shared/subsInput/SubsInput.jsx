import {useState} from 'react'
import {useNavigate} from "react-router-dom"
import {FaTelegramPlane} from "react-icons/fa"
import { v4 as uuidv4 } from 'uuid';
import { doc, setDoc } from "firebase/firestore"; 
import db from "../../../db/firebase.config"

function SubsInput(){
    
  const [email, setEmail] = useState("")
  const navigate = useNavigate()

  async function sendData(obj){
    
    await setDoc(doc(db, "subscription-list", uuidv4()), {
      email: obj
    });

  }

  function handleClick(){
    sendData(email).then(setEmail("")).then(navigate('thanks-subs'))
  }

    return(
        <div className='flex justify-start items-center'> 
              <input 
              type='email'
              className='border border-gray-400 rounded-l-md h-10 pl-2 focus:outline-none ' 
              placeholder='Enter your e-mail'
              value={email} 
              onChange={e => setEmail(e.target.value)} 
              />
              <button type='button' className='h-10  bg-cyan-400 pl-1 pr-2 rounded-r-md border-y border-r border-gray-400 hover:bg-cyan-300 text-xl' onClick={() => handleClick()}>
                <FaTelegramPlane />
              </button>
          </div>
    )
}

export default SubsInput