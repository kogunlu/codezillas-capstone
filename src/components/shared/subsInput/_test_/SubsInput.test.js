import { BrowserRouter } from "react-router-dom"
import renderer from "react-test-renderer"
import SubsInput from "../SubsInput"


const mockedUsedNavigate = jest.fn()

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUsedNavigate
}))

it('renders correctly', async() => {
    const subsinputComponent = renderer
    .create(
        <BrowserRouter>
            <SubsInput />
        </BrowserRouter>
    )
    .toJSON()
        expect(subsinputComponent).toMatchSnapshot()
    
})