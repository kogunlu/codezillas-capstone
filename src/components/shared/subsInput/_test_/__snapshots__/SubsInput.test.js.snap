// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex justify-start items-center"
>
  <input
    className="border border-gray-400 rounded-l-md h-10 pl-2 focus:outline-none "
    onChange={[Function]}
    placeholder="Enter your e-mail"
    type="email"
    value=""
  />
  <button
    className="h-10  bg-cyan-400 pl-1 pr-2 rounded-r-md border-y border-r border-gray-400 hover:bg-cyan-300 text-xl"
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M446.7 98.6l-67.6 318.8c-5.1 22.5-18.4 28.1-37.3 17.5l-103-75.9-49.7 47.8c-5.5 5.5-10.1 10.1-20.7 10.1l7.4-104.9 190.9-172.5c8.3-7.4-1.8-11.5-12.9-4.1L117.8 284 16.2 252.2c-22.1-6.9-22.5-22.1 4.6-32.7L418.2 66.4c18.4-6.9 34.5 4.1 28.5 32.2z"
      />
    </svg>
  </button>
</div>
`;
